### PR TITLE
Istio 1.21 Support

### PR DESCRIPTION
**Pull request by sheidkamp**
# Description
Changes to support Istio 1.21.2

## Code changes
Removed the regexes used by the [ Istio sidecar matcher](https://github.com/solo-io/gloo/pull/9500/files#diff-7ad53da28a038b19fc9e8626916092b1497f431ec9296906a3a6526ff8ad54d9L19) that are used to support versions 1.6 and below of Isito. All supported versions of Istio use the same sidecar, and the code now reflects this.

## CI changes
Update `setup-kind.sh` and kube2e test artifacts to use Istio 1.21.2.

## Docs changes
Updated supported versions

# Context
[Istio Change Notes review and planning doc](https://soloio.slab.com/posts/istio-1-21-for-gloo-edge-12jrp5yn)

## Interesting decisions
Accepted the default behavior of [adding startupProbe to sidecar by default](https://istio.io/latest/news/releases/1.20.x/announcing-1.20/upgrade-notes/#startupprobe-added-to-sidecar-by-default). Because this change affects the startup of the pods, we are covering the behavior in our e2e tests. Because the change is limited to a container in the pod, the pod's behavior as a whole will not appear to have changed.

## Testing steps

We are relying on the existing CI to validate that expected functionality is not broken.

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by sheidkamp** [comment]
/kick
```
ERROR: error fetching storage source: generic::unknown: retry budget exhausted (3 attempts): fetching gcs source: fetching object from gcs: source fetch container exited with non-zero status: 1
```

**Review by Nadine2016 — APPROVED** [review]

**Review by nfuden — COMMENTED** [review]
LGTM but would like Nina to have a look before it merges

File: docs/content/reference/support.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ Gloo Edge Enterprise offers `n-3` patching support for bug and critical security
 
 | Gloo Edge | Kubernetes | Envoy | Helm | Istio`†`    |
 |------|----------|---------|--------|-------------|
+| 1.17.x | 1.24 - 1.28 | v3 xDS API | >= 3.12 | 1.16 - 1.21 |
```
Comment (npolshakova) [inline docs/content/reference/support.md:15]: Why is the lower supported version 1.16 here instead of 1.15?

Comment (sheidkamp) [reply docs/content/reference/support.md:15]: I was going off the supported versions from this ticket: https://github.com/solo-io/solo-projects/issues/5752 happy to update.

**Review by npolshakova — APPROVED** [review]

File: docs/content/reference/support.md (hunk 1 above)
Comment (sheidkamp) [reply docs/content/reference/support.md:15]: Discussion: https://solo-io-corp.slack.com/archives/C03MFATU265/p1716567997214099?thread_ts=1716558520.686479&cid=C03MFATU265

**Comment by sheidkamp** [comment]
dozer?

**Pull request merged**